crates/bidiff/src/lib.rs

```diff
@@ -296,15 +297,39 @@ impl<'a> Iterator for BsdiffIterator<'a> {
     }
 }
 
+/// Parameters used when creating diffs
 pub struct DiffParams {
-    // Number of partitions to use for suffix sorting.
-    // Increase this number increases parallelism but produces slightly worse patches.
-    pub sort_partitions: usize,
-
-    // Size of chunks to use for scanning. When None, treat the
-    // input as a single chunk. Smaller chunks increase parallelism but
-    // produce slightly worse patches.
-    pub scan_chunk_size: Option<usize>,
+    sort_partitions: usize,
+    scan_chunk_size: Option<usize>,
+}
+
+impl DiffParams {
+    /// Construct new diff params and check validity
+    ///
+    /// # Parameters
+    ///
+    /// - `sort_partitions`: Number of partitions to use for suffix sorting.
+    ///   Increase this number increases parallelism but produces slightly worse
+    ///   patches. Needs to be at least 1.
+    /// - `scan_chunk_size`: Size of chunks to use for scanning. When `None`, treat
+    ///   the input as a single chunk. Smaller chunks increase parallelism but
+    ///   produce slightly worse patches. When `Some`, it needs to be at least 1.
+    pub fn new(
+        sort_partitions: usize,
+        scan_chunk_size: Option<usize>,
+    ) -> Result<Self, Box<dyn Error + Send + Sync + 'static>> {
+        if sort_partitions < 1 {
+            return Err("number of sort partitions cannot be less than 1".into());
+        }
+        if scan_chunk_size.filter(|s| *s < 1).is_some() {
```

Comment:
seems like a gratuitous use of filter just to avoid an if-let+if combo lol

Reply:
Indentation is the enemy!

Want me to change it?

Reply:
Nah it's fine